gluon/scheduler.py

```diff
@@ -817,11 +825,14 @@ def pop_task(self, db):
         if not task.prevent_drift:
             next_run_time = task.last_run_time + datetime.timedelta(
                 seconds=task.period
-                )
-        else:
-            next_run_time = task.start_time + datetime.timedelta(
-                seconds=task.period * times_run
             )
+        else:
+            # calc next_run_time based on available slots
+            # see #1191
+            secondspassed = self.total_seconds(now - task.start_time)
+            steps = secondspassed // task.period + 1
+            next_run_time += datetime.timedelta(seconds=task.period * steps)
```

Comment:
Nope. next_run_time hasn't even been used before so this will always cause an exception.

Should be this

`next_run_time = task.start_time + datetime.timedelta(seconds=task.period * steps)`